### PR TITLE
Fix broken tests following instance key changes

### PR DESCRIPTION
**Pull request by teh-cmc**
### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6130?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6130?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6130)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_data_store/src/store_gc.rs [hunk 1]
```diff
@@ -731,6 +731,7 @@ impl IndexedBucketInner {
                     } else {
                         let mut d = StoreDiff::deletion(removed_row_id, entity_path.clone());
                         d.at_timestamp(timeline, time);
+                        d.insert(cell);
```
Comment (teh-cmc) [inline crates/re_data_store/src/store_gc.rs:734]: bruh that was nasty

**Comment by teh-cmc** [comment]
@rerun-bot full-check

File: crates/re_data_store/tests/snapshots/memory_test__scalars_on_one_timeline.snap [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
 source: crates/re_data_store/tests/memory_test.rs
-assertion_line: 96
 expression: "[format!(\"{NUM_SCALARS} scalars\"),\n        format!(\"{} in total\", re_format::format_bytes(total_mem_use as _)),\n        format!(\"{} per row\",\n            re_format::format_bytes(total_mem_use as f64 / NUM_SCALARS as\n            f64))]"
 ---
 [
     "1048576 scalars",
-    "936 MiB in total",
-    "936 B per row",
+    "912 MiB in total",
+    "912 B per row",
```
Comment (emilk) [inline crates/re_data_store/tests/snapshots/memory_test__scalars_on_one_timeline.snap:8]: That's one small step in the right direction 😆

**Review by emilk — APPROVED** [review]

File: crates/re_data_store/tests/snapshots/memory_test__scalars_on_one_timeline.snap (hunk 2 above)
Comment (teh-cmc) [reply crates/re_data_store/tests/snapshots/memory_test__scalars_on_one_timeline.snap:8]: :disappointed:

**Pull request merged**